File: 백준/Silver/12018. Yonsei TOTO/Yonsei TOTO.py
import sys
input = sys.stdin.readline

n, m = list(map(int, input().strip().split()))
li = []
for _ in range(n):
    P, L = list(map(int, input().strip().split()))
    point = list(map(int, input().strip().split()))
    point.sort(reverse=True)

    # 마지막에 문 닫고 들온 놈 뚧으려면 내가 걔보다 point 더 써야 한다
    if P >= L:
        li.append(point[L-1])
    else:
        li.append(1)


# 25, 20, 36, 14
li.sort()
cnt = 0
for i in li:
    if m-i >= 0:
        cnt += 1
        m -= i
print(cnt)

File: 백준/Silver/12018. Yonsei TOTO/Yonsei TOTO.java
import java.io.IOException;
import java.io.BufferedReader;
import java.io.InputStreamReader;

import java.util.Arrays;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		String[] N_mileage = br.readLine().split(" ");
		int N = Integer.parseInt(N_mileage[0]);
		int mileage = Integer.parseInt(N_mileage[1]);
		
		int[] arr = new int[N];
		for (int i = 0; i < N; i++) {
			String[] P_L = br.readLine().split(" ");
			String[] score_arr = br.readLine().split(" ");
			int P = Integer.parseInt(P_L[0]);
			int L = Integer.parseInt(P_L[1]);
			int[] score = new int[P];
			for (int j = 0; j < P; j++) {
				score[j] = Integer.parseInt(score_arr[j]);
			}
			Arrays.sort(score);
			
			if (P >= L) {
				arr[i] = score[P-L];
			}
			else if (P < L) {
				arr[i] = 1;
			}
		}
		Arrays.sort(arr);
		
		int cnt = 0;
		for (int i = 0; i < N; i++) {
			if (mileage - arr[i] >= 0) {
				mileage -= arr[i];
				cnt++;
			}
		}
		System.out.println(cnt);
		
	}
}